workspaces/ui-v2/src/store/documentationEdit/slice.ts

```diff
@@ -92,7 +92,8 @@ const documentationEditSlice = createSlice({
       state.removedEndpoints = newRemovedEndpoints;
     },
     removeField: (state, action: PayloadAction<{ fieldId: string }>) => {
-      // TODO FLEB filter any other removed fields that is now a child of the newly added field.
+      // @GOTCHA - selecting a field, and then selecting a parent of this field (e.g. an object that contains this removed field)
```

Comment:
I decided I'm not going to fix this as it's not trivial to fix with how we've modeled the store. This slice only has access to documentationEdits - and this logic needs to look at the shape tree. The handling should be in this action. To solve this we could:
- include the current store shapeMap in the action payload
- Restructure the redux store

Since this is only going to generate more commands, but the commands should be valid regardless, this is fine to keep in.

Reply:
Minimising the amount of commands / resulting events is definitely an optimisation we can skip for now!